src/rdiff_backup/rpath.py

```diff
@@ -997,8 +1000,22 @@ def get_temp_rpath(self, sibling=False):
             if self.__class__._temp_file_index > 100000000:
                 log.Log("Warning: Resetting tempfile index", 2)
                 self.__class__._temp_file_index = 0
-            tf = self.append('rdiff-backup.tmp.{index:d}'.format(
-                             index=self.__class__._temp_file_index))
+
+            if tempfile.tempdir and (shutil.disk_usage(self.path).free == 0):
+                # here to catch the bytes string passed via cli opts
+                if type(tempfile.tempdir) == bytes:
+                    tempdir = tempfile.tempdir.decode()
+                else:
+                    tempdir = tempfile.tempdir
+            else:
+                tempdir = None
+
+            if not tempdir:
+                tf = self.append('rdiff-backup.tmp.{index:d}'.format(
+                                 index=self.__class__._temp_file_index))
+            else:
+                tf = self.append(os.path.join(tempdir, 'rdiff-backup.tmp.{index:d}'.format(
```

Comment:
This line sounds really wrong: if tempdir is `/var/tmp`, this line will _append_ it to the current path, i.e. `/mnt/mybackup` would then give `/var/tmp/mnt/mybackup/rdiff-backup.tmp.123`. This is surely not what you want.

Rather create a completely new RPath object, probably something like `tf = RPath(self.conn, tempfile.tempdir, (indexed_file_name))`. Notice that I assume `indexed_file_name = os.encode("rdiff-backup.tmp.{index:d}".format(index=self.__class__._temp_file_index)))` (it must be bytes, the type of tempfile.tempdir becomes irrelevant).

Reply:
Instead of building a completely new RPath object, wouldn't it be enough to avoid appending and let tf be composed of the two joint directories? Went that way with https://github.com/rdiff-backup/rdiff-backup/pull/545/commits/d48390b80cf4de90f79926b8e3a3d4cdd4a2efea, please let me know if that's not desired.